core-bundle/contao/modules/ModuleLogin.php

```diff
@@ -74,28 +74,25 @@ public function generate()
 		{
 			$this->targetPath = base64_decode($request->request->get('_target_path'));
 		}
-		elseif ($request && $this->redirectBack)
+		if ($request?->query->has('redirect'))
```

Comment:
this **must** remain an `elseif` because we always want to prefer the POST data imho

Reply:
Of course. That's how I actually wanted to do it. 🙈

Fixed in 49a2e58576e5e592b2bae9806dcdc9f668be4bd6.